### ISSUE TITLE
Implement/refactor GA tracking events for remote site notifications

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#4056]
## Feature Description

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by asvinb** [comment]
@felixarntz I was chatting with @aaemnnosttv about the current ticket and right now we don't have a way to distinguish between server and client notifications since both sets of notifications get merged when using the `getNotifications` selector.
https://github.com/google/site-kit-wp/blob/43e3866a3a5a523f12ebb27938cc178a6e281a02/assets/js/googlesitekit/data/create-notifications-store.js#L225-L228

We can add a new `source` property to notifications or add a new selector, `isServerNotification` with an `id` as parameter which will tell whether the notification is a server or client one.

What do you think?

**Comment by felixarntz** [comment]
@asvinb @aaemnnosttv For which client notifications is this store actually currently used, if any? We may just be okay not caring about where exactly a notification comes from and then for example using `dashboard_remote-site-notification` instead of `dashboard_site-notification` as event category. The notification ID would implicitly tell us where it is from, since we know which ones are remote-controlled.

**Comment by aaemnnosttv** [comment]
> For which client notifications is this store actually currently used, if any?

@felixarntz I don't think we use it for any client notifications right now, but the functionality is there for it, so we didn't want to introduce assumptions that might cause problems later 😄

**Comment by asvinb** [comment]
> @asvinb @aaemnnosttv For which client notifications is this store actually currently used, if any? We may just be okay not caring about where exactly a notification comes from and then for example using `dashboard_remote-site-notification` instead of `dashboard_site-notification` as event category. The notification ID would implicitly tell us where it is from, since we know which ones are remote-controlled.

@felixarntz Did you mean to use `dashboard_site-notification` instead of `dashboard_remote-site-notification` 😅 
I've updated the IB, can you take a look. Also, we can add the tracking calls for when the CTA is clicked and the notification is dismissed in several places, for e.g in `assets/js/components/legacy-notifications/site/mark-notification.js` or `assets/js/components/legacy-notifications/DashboardCoreSiteAlerts.js` but like I said in the IB, I suggest we keep all the tracking calls in `Notification` so it's easier to manage and have the tracking calls in one file as opposed to having it spread in multiple files.

**Comment by felixarntz** [comment]
@asvinb Oh indeed, I meant to use `dashboard_site-notification`, messed up the order in my comment. Can you update the IB to use that category? Other than that, the IB looks good. It was just a bit confusing to read only as the last bullet point that the notification ID must be present in each event - which is correct, but above you also mention it for the `view_notification` event, but then don't mention it for the others. Maybe we should remove it from there or state it explicitly for each event - I just want to prevent that when this gets implemented the notification ID as event label gets missed in one of those events.

**Comment by asvinb** [comment]
Thanks @felixarntz . IB updated as per your comments!

**Comment by felixarntz** [comment]
@asvinb Thanks - IB ✅

**Comment by kuasha420** [comment]
@eugene-manuilov AC has been updated by @felixarntz. It is ready for another review.

**Comment by ivankruchkoff** [comment]
QA ✅ 

Confirmed, event tracking fired correctly:

```
https://www.google-analytics.com/collect?v=1&_v=j93&a=139XXXXXXX&t=event&_s=3&dl=http://sitekit.10uplabs.com/wp-admin/admin.php?page=googlesitekit-dashboard&ul=en-us&de=UTF-8&dt=Site Kit by Google Dashboard ‹ sitekit — WordPress&sd=24-bit&sr=1440x900&vp=1425x256&je=0&ec=dashboard_site-notification&ea=view_notification&el=googlesitekit-error&_u=aADAAUABAAAAAC~&jid=&gjid=&cid=863XXXXXX.XXXX6XXXX0&tid=UA-XXXXXXXXX-3&_gid=2XXXXXXXXX.XXXXXXXXXX&gtm=2XXXX0&cd1=http://sitekit.10uplabs.com&cd2=true&cd3=XXXXXXXXXXXXXXXXX&cd4=1.43.0&cd5=&z=22697XXXX
```
